trin-history/src/jsonrpc.rs

```diff
@@ -22,6 +28,66 @@ impl HistoryRequestHandler {
     pub async fn handle_client_queries(mut self) {
         while let Some(request) = self.history_rx.recv().await {
             match request.endpoint {
+                HistoryEndpoint::GetBlockByHash => {
```

Comment:
As stated above, I don't think we should add custom endpoints for overlay networks, and instead we should implement this functionality with existing messages like `FindContent`.